### ISSUE TITLE
mlir-opt doesn't link: `undefined reference to 'mlir::test::registerTestLowerToNVVM()'`

### ISSUE THREAD

**Issue by sylvestre** [llvm/llvm-project#63913]
Started recently on linux with shared libraries:
```

FAILED: bin/mlir-opt 
: && /build/source/build-llvm/./bin/clang++ -fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -ffile-prefix-map=/build/source/build-llvm/tools/clang/stage2-bins=build-llvm/tools/clang/stage2-bins -ffile-prefix-map=/build/source/= -no-canonical-prefixes -Werror=mismatched-tags -O2 -DNDEBUG -g1 -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id -fuse-ld=gold   -Wl,--export-dynamic tools/mlir/tools/mlir-opt/CMakeFiles/mlir-opt.dir/mlir-opt.cpp.o -o bin/mlir-opt  -Wl,-rpath,"\$ORIGIN/../lib:/build/source/build-llvm/tools/clang/stage2-bins/lib:"  lib/libMLIRAffineAnalysis.a  lib/libMLIRAffineDialect.a  lib/libMLIRAffineTransforms.a  lib/libMLIRAffineTransformOps.a  lib/libMLIRAffineUtils.a  lib/libMLIRAMDGPUDialect.a  lib/libMLIRAMDGPUTransforms.a  lib/libMLIRAMDGPUUtils.a  lib/libMLIRAMXDialect.a  lib/libMLIRAMXTransforms.a  lib/libMLIRArithDialect.a  lib/libMLIRArithValueBoundsOpInterfaceImpl.a  lib/libMLIRArithTransforms.a  lib/libMLIRArithUtils.a  lib/libMLIRArmNeonDialect.a  lib/libMLIRArmSMEDialect.a  lib/libMLIRArmSMETransforms.a  lib/libMLIRArmSVEDialect.a  lib/libMLIRArmSVETransforms.a  lib/libMLIRAsyncDialect.a  lib/libMLIRAsyncTransforms.a  lib/libMLIRBufferizationDialect.a  lib/libMLIRBufferizationTransformOps.a  lib/libMLIRBufferizationTransforms.a  lib/libMLIRComplexDialect.a  lib/libMLIRControlFlowDialect.a  lib/libMLIRDLTIDialect.a  lib/libMLIREmitCDialect.a  lib/libMLIRFuncDialect.a  lib/libMLIRFuncTransforms.a  lib/libMLIRGPUDialect.a  lib/libMLIRGPUTransforms.a  lib/libMLIRGPUTransformOps.a  lib/libMLIRIndexDialect.a  lib/libMLIRIRDL.a  lib/libMLIRLinalgDialect.a  lib/libMLIRLinalgTransformOps.a  lib/libMLIRLinalgTransforms.a  lib/libMLIRLinalgUtils.a  lib/libMLIRLLVMIRTransforms.a  lib/libMLIRLLVMDialect.a  lib/libMLIRNVVMDialect.a  lib/libMLIRROCDLDialect.a  lib/libMLIRMathDialect.a  lib/libMLIRMathTransforms.a  lib/libMLIRMemRefDialect.a  lib/libMLIRMemRefTransformOps.a  lib/libMLIRMemRefTransforms.a  lib/libMLIRMemRefUtils.a  lib/libMLIRMLProgramDialect.a  lib/libMLIRNVGPUDialect.a  lib/libMLIRNVGPUUtils.a  lib/libMLIRNVGPUTransformOps.a  lib/libMLIRNVGPUTransforms.a  lib/libMLIROpenACCDialect.a  lib/libMLIROpenMPDialect.a  lib/libMLIRPDLDialect.a  lib/libMLIRPDLInterpDialect.a  lib/libMLIRQuantDialect.a  lib/libMLIRQuantUtils.a  lib/libMLIRSCFDialect.a  lib/libMLIRSCFTransformOps.a  lib/libMLIRSCFTransforms.a  lib/libMLIRSCFUtils.a  lib/libMLIRShapeDialect.a  lib/libMLIRShapeOpsTransforms.a  lib/libMLIRSparseTensorDialect.a  lib/libMLIRSparseTensorTransforms.a  lib/libMLIRSparseTensorPipelines.a  lib/libMLIRSparseTensorUtils.a  lib/libMLIRSPIRVDialect.a  lib/libMLIRSPIRVModuleCombiner.a  lib/libMLIRSPIRVConversion.a  lib/libMLIRSPIRVTransforms.a  lib/libMLIRSPIRVUtils.a  lib/libMLIRTensorDialect.a  lib/libMLIRTensorInferTypeOpInterfaceImpl.a  lib/libMLIRTensorTilingInterfaceImpl.a  lib/libMLIRTensorTransforms.a  lib/libMLIRTensorTransformOps.a  lib/libMLIRTensorUtils.a  lib/libMLIRTosaDialect.a  lib/libMLIRTosaTransforms.a  lib/libMLIRTransformDialect.a  lib/libMLIRTransformPDLExtension.a  lib/libMLIRTransformDialectTransforms.a  lib/libMLIRTransformDialectUtils.a  lib/libMLIRVectorDialect.a  lib/libMLIRVectorTransforms.a  lib/libMLIRVectorTransformOps.a  lib/libMLIRVectorUtils.a  lib/libMLIRX86VectorDialect.a  lib/libMLIRX86VectorTransforms.a  lib/libMLIRTestDynDialect.a  lib/libMLIRTosaTestPasses.a  lib/libMLIRAffineToStandard.a  lib/libMLIRAMDGPUToROCDL.a  lib/libMLIRArithAttrToLLVMConversion.a  lib/libMLIRArithToLLVM.a  lib/libMLIRArithToSPIRV.a  lib/libMLIRArmNeon2dToIntr.a  lib/libMLIRAsyncToLLVM.a  lib/libMLIRBufferizationToMemRef.a  lib/libMLIRComplexToLLVM.a  lib/libMLIRComplexToLibm.a  lib/libMLIRComplexToSPIRV.a  lib/libMLIRComplexToStandard.a  lib/libMLIRControlFlowToLLVM.a  lib/libMLIRControlFlowToSPIRV.a  lib/libMLIRFuncToLLVM.a  lib/libMLIRFuncToSPIRV.a  lib/libMLIRGPUToGPURuntimeTransforms.a  lib/libMLIRGPUToNVVMTransforms.a  lib/libMLIRGPUToROCDLTransforms.a  lib/libMLIRGPUToSPIRV.a  lib/libMLIRGPUToVulkanTransforms.a  lib/libMLIRIndexToLLVM.a  lib/libMLIRLinalgToLLVM.a  lib/libMLIRLinalgToStandard.a  lib/libMLIRLLVMCommonConversion.a  lib/libMLIRMathToFuncs.a  lib/libMLIRMathToLibm.a  lib/libMLIRMathToLLVM.a  lib/libMLIRMathToSPIRV.a  lib/libMLIRMemRefToLLVM.a  lib/libMLIRMemRefToSPIRV.a  lib/libMLIRNVGPUToNVVM.a  lib/libMLIRNVVMToLLVM.a  lib/libMLIROpenACCToSCF.a  lib/libMLIROpenMPToLLVM.a  lib/libMLIRPDLToPDLInterp.a  lib/libMLIRReconcileUnrealizedCasts.a  lib/libMLIRSCFToControlFlow.a  lib/libMLIRSCFToGPU.a  lib/libMLIRSCFToOpenMP.a  lib/libMLIRSCFToSPIRV.a  lib/libMLIRShapeToStandard.a  lib/libMLIRSPIRVToLLVM.a  lib/libMLIRTensorToLinalg.a  lib/libMLIRTensorToSPIRV.a  lib/libMLIRTosaToArith.a  lib/libMLIRTosaToLinalg.a  lib/libMLIRTosaToSCF.a  lib/libMLIRTosaToTensor.a  lib/libMLIRVectorToLLVM.a  lib/libMLIRVectorToGPU.a  lib/libMLIRVectorToSCF.a  lib/libMLIRVectorToSPIRV.a  lib/libMLIRFuncInlinerExtension.a  lib/libMLIRFuncAllExtensions.a  lib/libMLIRTestFuncToLLVM.a  lib/libMLIRAffineTransformsTestPasses.a  lib/libMLIRArithTestPasses.a  lib/libMLIRBufferizationTestPasses.a  lib/libMLIRControlFlowTestPasses.a  lib/libMLIRDLTITestPasses.a  lib/libMLIRFuncTestPasses.a  lib/libMLIRGPUTestPasses.a  lib/libMLIRLinalgTestPasses.a  lib/libMLIRLoopLikeInterfaceTestPasses.a  lib/libMLIRMathTestPasses.a  lib/libMLIRMemRefTestPasses.a  lib/libMLIRNVGPUTestPasses.a  lib/libMLIRSCFTestPasses.a  lib/libMLIRShapeTestPasses.a  lib/libMLIRSPIRVTestPasses.a  lib/libMLIRTensorTestPasses.a  lib/libMLIRTestAnalysis.a  lib/libMLIRTestDialect.a  lib/libMLIRTestDynDialect.a  lib/libMLIRTestIR.a  lib/libMLIRTestOneToNTypeConversionPass.a  lib/libMLIRTestPass.a  lib/libMLIRTestPDLL.a  lib/libMLIRTestReducer.a  lib/libMLIRTestRewrite.a  lib/libMLIRTestTransformDialect.a  lib/libMLIRTestTransforms.a  lib/libMLIRTilingInterfaceTestPasses.a  lib/libMLIRVectorTestPasses.a  lib/libMLIRTestVectorToSPIRV.a  lib/libMLIRLLVMTestPasses.a  lib/libMLIRAffineAnalysis.a  lib/libMLIRAnalysis.a  lib/libMLIRCastInterfaces.a  lib/libMLIRDialect.a  lib/libMLIROptLib.a  lib/libMLIRParser.a  lib/libMLIRPass.a  lib/libMLIRTransforms.a  lib/libMLIRTransformUtils.a  lib/libMLIRSupport.a  lib/libMLIRIR.a  lib/libMLIRSparseTensorTransforms.a  lib/libMLIRSparseTensorUtils.a  lib/libMLIRAMDGPUToROCDL.a  lib/libMLIRAMDGPUUtils.a  lib/libMLIRAMDGPUDialect.a  lib/libMLIRMemRefToSPIRV.a  lib/libMLIRSPIRVSerialization.a  lib/libMLIRSPIRVBinaryUtils.a  lib/libMLIROpenACCDialect.a  lib/libMLIROpenMPDialect.a  lib/libMLIRSPIRVUtils.a  lib/libMLIRArithToSPIRV.a  lib/libMLIRFuncToSPIRV.a  lib/libMLIRTosaTransforms.a  lib/libMLIRTosaDialect.a  lib/libMLIRQuantUtils.a  lib/libMLIRQuantDialect.a  lib/libMLIRNVGPUUtils.a  lib/libMLIRFuncInlinerExtension.a  lib/libMLIRGPUToGPURuntimeTransforms.a  lib/libMLIRAsyncToLLVM.a  lib/libMLIRNVVMToLLVMIRTranslation.a  lib/libMLIRGPUTransforms.a  lib/libMLIRAsyncDialect.a  lib/libMLIRBuiltinToLLVMIRTranslation.a  lib/libMLIRExecutionEngineUtils.a  lib/libMLIRGPUToLLVMIRTranslation.a  lib/libMLIRROCDLToLLVMIRTranslation.a  lib/libMLIRROCDLDialect.a  lib/libMLIRLLVMToLLVMIRTranslation.a  lib/libMLIRMathTransforms.a  lib/libMLIRNVGPUTransforms.a  lib/libMLIRShapeOpsTransforms.a  lib/libMLIRShapeDialect.a  lib/libMLIRSPIRVModuleCombiner.a  lib/libMLIRTransformPDLExtension.a  lib/libMLIRTransformDialectTransforms.a  lib/libMLIRTransformDialect.a  lib/libMLIRTransformDialectUtils.a  lib/libMLIRTestDialect.a  lib/libMLIRDerivedAttributeOpInterface.a  lib/libMLIRReduce.a  lib/libMLIRVectorToSPIRV.a  lib/libMLIRSPIRVConversion.a  lib/libMLIRSPIRVDialect.a  lib/libMLIRLinalgTransforms.a  lib/libMLIRSCFTransforms.a  lib/libMLIRTensorTilingInterfaceImpl.a  lib/libMLIRLinalgUtils.a  lib/libMLIRTensorTransforms.a  lib/libMLIRTensorUtils.a  lib/libMLIRMemRefTransforms.a  lib/libMLIRAffineTransforms.a  lib/libMLIRSCFUtils.a  lib/libMLIRArithTransforms.a  lib/libMLIRFuncTransforms.a  lib/libMLIRNVGPUDialect.a  lib/libMLIRFuncToLLVM.a  lib/libMLIRArithToLLVM.a  lib/libMLIRControlFlowToLLVM.a  lib/libMLIRIndexToLLVM.a  lib/libMLIRIndexDialect.a  lib/libMLIRLinalgToLLVM.a  lib/libMLIRAffineToStandard.a  lib/libMLIRMathToLLVM.a  lib/libMLIRArithAttrToLLVMConversion.a  lib/libMLIRMemRefToLLVM.a  lib/libMLIRReconcileUnrealizedCasts.a  lib/libMLIRSCFToControlFlow.a  lib/libMLIRVectorToLLVM.a  lib/libMLIRAMXTransforms.a  lib/libMLIRAMXDialect.a  lib/libMLIRArmNeonDialect.a  lib/libMLIRArmSMETransforms.a  lib/libMLIRArmSMEDialect.a  lib/libMLIRArmSVETransforms.a  lib/libMLIRArmSVEDialect.a  lib/libMLIRX86VectorTransforms.a  lib/libMLIRX86VectorDialect.a  lib/libMLIRLLVMCommonConversion.a  lib/libMLIRTargetLLVMIRExport.a  lib/libMLIRLLVMIRTransforms.a  lib/libMLIRNVVMDialect.a  lib/libMLIRTranslateLib.a  lib/libMLIRVectorToSCF.a  lib/libMLIRLLVMDialect.a  lib/libMLIRVectorTransforms.a  lib/libMLIRAffineUtils.a  lib/libMLIRBufferizationTransforms.a  lib/libMLIRGPUDialect.a  lib/libMLIRDLTIDialect.a  lib/libMLIRLinalgDialect.a  lib/libMLIRMathDialect.a  lib/libMLIRTilingInterface.a  lib/libMLIRMemRefUtils.a  lib/libMLIRVectorUtils.a  lib/libMLIRAffineAnalysis.a  lib/libMLIRSCFDialect.a  lib/libMLIRBufferizationDialect.a  lib/libMLIRSparseTensorDialect.a  lib/libMLIRControlFlowDialect.a  lib/libMLIRFuncDialect.a  lib/libMLIRVectorDialect.a  lib/libMLIRTensorDialect.a  lib/libMLIRAffineDialect.a  lib/libMLIRMemRefDialect.a  lib/libMLIRComplexDialect.a  lib/libMLIRShapedOpInterfaces.a  lib/libMLIRValueBoundsOpInterface.a  lib/libMLIRParallelCombiningOpInterface.a  lib/libMLIRDestinationStyleOpInterface.a  lib/libMLIRMaskableOpInterface.a  lib/libMLIRMaskingOpInterface.a  lib/libMLIRDialectUtils.a  lib/libMLIRArithUtils.a  lib/libMLIRArithDialect.a  lib/libMLIRCastInterfaces.a  lib/libMLIRInferIntRangeCommon.a  lib/libMLIRVectorInterfaces.a  lib/libMLIRTransforms.a  lib/libMLIRTransformUtils.a  lib/libMLIRRewrite.a  lib/libMLIRPDLToPDLInterp.a  lib/libMLIRPDLInterpDialect.a  lib/libMLIRPDLDialect.a  lib/libMLIRMemorySlotInterfaces.a  lib/libMLIRCopyOpInterface.a  lib/libMLIRRuntimeVerifiableOpInterface.a  lib/libMLIRIRDL.a  lib/libMLIRDialect.a  lib/libMLIRParser.a  lib/libMLIRBytecodeReader.a  lib/libMLIRAsmParser.a  lib/libMLIRBytecodeWriter.a  lib/libMLIRBytecodeOpInterface.a  lib/libMLIRDebug.a  lib/libMLIRObservers.a  lib/libMLIRPluginsLib.a  lib/libMLIRPass.a  lib/libMLIRAnalysis.a  lib/libMLIRCallInterfaces.a  lib/libMLIRControlFlowInterfaces.a  lib/libMLIRInferTypeOpInterface.a  lib/libMLIRSideEffectInterfaces.a  lib/libMLIRPresburger.a  lib/libMLIRLoopLikeInterface.a  lib/libMLIRViewLikeInterface.a  lib/libMLIRInferIntRangeInterface.a  lib/libMLIRDataLayoutInterfaces.a  lib/libMLIRIR.a  lib/libMLIRSupport.a  lib/libLLVM-17.so.1 && :
build-llvm/tools/clang/stage2-bins/mlir/tools/mlir-opt/mlir-opt.cpp:203: error: undefined reference to 'mlir::test::registerTestLowerToNVVM()'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by joker-eph** [comment]
What's your cmake invocation? (I have a bot building with shared libs = ON)

**Comment by sylvestre** [comment]
it fails on stage 2

`cd /build/source/build-llvm/tools/clang/stage2-bins && /usr/bin/cmake -DCMAKE_INSTALL_PREFIX=/usr/lib/llvm-17 -DCMAKE_INSTALL_PREFIX=/usr/lib/llvm-17 -DCMAKE_SUPPRESS_REGENERATION=ON -DENABLE_LINKER_BUILD_ID=ON -DLLVM_ENABLE_PIC=ON -DLLVM_ENABLE_RTTI=ON -DLLVM_INCLUDE_GO_TESTS=OFF -DLLVM_USE_RELATIVE_PATHS_IN_FILES=ON -DCLANG_DEFAULT_LINKER=ld -DCLANG_DEFAULT_CXX_STDLIB= -DCLANG_DEFAULT_RTLIB=libgcc -DCOMPILER_RT_USE_LIBCXX=OFF -DCOMPILER_RT_USE_BUILTINS_LIBRARY=ON -DCOMPILER_RT_INCLUDE_TESTS=OFF -DLIBUNWIND_USE_COMPILER_RT=ON -DLIBCXXABI_ENABLE_EXCEPTIONS=ON -DLIBCXXABI_USE_COMPILER_RT=ON -DLIBCXX_USE_COMPILER_RT=ON -DLIBCXX_ENABLE_EXCEPTIONS=ON -DLIBCXX_ENABLE_STATIC_ABI_LIBRARY=ON -DLIBCXX_STATICALLY_LINK_ABI_IN_STATIC_LIBRARY=ON -DLIBCXX_STATICALLY_LINK_ABI_IN_SHARED_LIBRARY=OFF -DLIBOMP_ENABLE_RTTI=OFF -DLLVM_TEMPORARILY_ALLOW_OLD_TOOLCHAIN=ON -DCLANG_FORCE_MATCHING_LIBCLANG_SOVERSION=ON -DCMAKE_C_COMPILER_LAUNCHER=/opt/sccache//sccache -DCMAKE_CXX_COMPILER_LAUNCHER=/opt/sccache//sccache -DPACKAGE_VERSION=17.0.0 -DPACKAGE_VENDOR=Ubuntu -DLLVM_VERSION_MAJOR=17 -DLLVM_VERSION_MINOR=0 -DLLVM_VERSION_PATCH=0 -DCLANG_VERSION_MAJOR=17 -DCLANG_VERSION_MINOR=0 -DCLANG_VERSION_PATCHLEVEL=0 -DCLANG_VENDOR=Ubuntu -DLLVM_VERSION_SUFFIX= -DLLVM_BINUTILS_INCDIR=/usr/include -DCLANG_REPOSITORY_STRING= -DCMAKE_MAKE_PROGRAM=/usr/bin/ninja -DCMAKE_BUILD_TYPE=Release "-DLLVM_ENABLE_PROJECTS=clang;clang-tools-extra;lld;cross-project-tests;mlir;bolt;flang;polly;openmp;lldb" "-DLLVM_ENABLE_RUNTIMES=compiler-rt;libcxx;libcxxabi;libunwind" "-DBUILTINS_CMAKE_ARGS=-DCMAKE_C_FLAGS=-fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2;-DCMAKE_CXX_FLAGS=-fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2;-DCMAKE_EXE_LINKER_FLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id;-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id;-DCMAKE_MODULE_LINKER_FLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id;-DCMAKE_BUILD_TYPE=RelWithDebInfo;-DCMAKE_C_FLAGS_RELWITHDEBINFO=-O2 -DNDEBUG -g1;-DCMAKE_CXX_FLAGS_RELWITHDEBINFO=-O2 -DNDEBUG -g1;-DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF" -DCLANGD_ENABLE_REMOTE=ON -DCLANG_DEFAULT_PIE_ON_LINUX=ON -DCLANG_LINK_CLANG_DYLIB=ON "-DCLANG_REPOSITORY_STRING=++20230717042521+4f95821f582e-1~exp1~20230717162637.198" -DCLANG_SYSTEMZ_DEFAULT_ARCH=z13 -DCMAKE_BUILD_TYPE=RelWithDebInfo "-DCMAKE_CXX_FLAGS=-fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2" "-DCMAKE_CXX_FLAGS_RELWITHDEBINFO=-O2 -DNDEBUG -g1" "-DCMAKE_C_FLAGS=-fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2" "-DCMAKE_C_FLAGS_RELWITHDEBINFO=-O2 -DNDEBUG -g1" "-DCMAKE_EXE_LINKER_FLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id" "-DCMAKE_MODULE_LINKER_FLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id" "-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id" -DLIBCLANG_LIBRARY_VERSION=1 -DLIBCXXABI_USE_LLVM_UNWINDER=ON -DLIBCXX_INSTALL_EXPERIMENTAL_LIBRARY=ON -DLIBOMP_LIBFLAGS=-lm -DLLDB_ENABLE_LUA=OFF -DLLVM_BINUTILS_INCDIR=/usr/include/ -DLLVM_ENABLE_CURL=ON -DLLVM_ENABLE_DUMP=ON -DLLVM_ENABLE_FFI=ON -DLLVM_ENABLE_LIBPFM=ON -DLLVM_ENABLE_SPHINX=ON -DLLVM_ENABLE_Z3_SOLVER=OFF "-DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=M68k;Xtensa" -DLLVM_INSTALL_UTILS=ON -DLLVM_LINK_LLVM_DYLIB=ON -DLLVM_POLLY_LINK_INTO_TOOLS=ON -DLLVM_USE_LINKER=gold -DLLVM_USE_PERF=ON -DLLVM_USE_RELATIVE_PATHS_IN_FILES=ON -DLLVM_VERSION_SUFFIX= -DMLIR_INSTALL_AGGREGATE_OBJECTS=OFF -DOPENMP_USE_LLVM_UNWINDER=ON -DPYTHON_EXECUTABLE=/usr/bin/python3 "-DRUNTIMES_CMAKE_ARGS=-DCMAKE_C_FLAGS=-fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2;-DCMAKE_CXX_FLAGS=-fstack-protector-strong -Wformat -Werror=format-security -Wno-unused-command-line-argument -Wdate-time -D_FORTIFY_SOURCE=2;-DCMAKE_EXE_LINKER_FLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id -L/build/source/build-llvm/lib;-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id -L/build/source/build-llvm/lib;-DCMAKE_MODULE_LINKER_FLAGS=-Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,--build-id -L/build/source/build-llvm/lib;-DCMAKE_BUILD_TYPE=RelWithDebInfo;-DCMAKE_C_FLAGS_RELWITHDEBINFO=-O2 -DNDEBUG -g1;-DCMAKE_CXX_FLAGS_RELWITHDEBINFO=-O2 -DNDEBUG -g1;-DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF" -DSPHINX_WARNINGS_AS_ERRORS=OFF -DLLVM_EXTERNAL_BOLT_SOURCE_DIR=/build/source/llvm/../bolt -DLLVM_EXTERNAL_CLANG_SOURCE_DIR=/build/source/llvm/../clang -DLLVM_EXTERNAL_CLANG_TOOLS_EXTRA_SOURCE_DIR=/build/source/llvm/../clang-tools-extra -DLLVM_EXTERNAL_CROSS_PROJECT_TESTS_SOURCE_DIR=/build/source/llvm/../cross-project-tests -DLLVM_EXTERNAL_FLANG_SOURCE_DIR=/build/source/llvm/../flang -DLLVM_EXTERNAL_LLDB_SOURCE_DIR=/build/source/llvm/../lldb -DLLVM_EXTERNAL_LLD_SOURCE_DIR=/build/source/llvm/../lld -DLLVM_EXTERNAL_MLIR_SOURCE_DIR=/build/source/llvm/../mlir -DLLVM_EXTERNAL_OPENMP_SOURCE_DIR=/build/source/llvm/../openmp -DLLVM_EXTERNAL_POLLY_SOURCE_DIR=/build/source/llvm/../polly -DCLANG_STAGE=stage2 -DCMAKE_CXX_COMPILER=/build/source/build-llvm/./bin/clang++ -DCMAKE_C_COMPILER=/build/source/build-llvm/./bin/clang -DCMAKE_ASM_COMPILER=/build/source/build-llvm/./bin/clang -DCMAKE_ASM_COMPILER_ID=Clang -GNinja /build/source/llvm && /usr/bin/cmake -E touch /build/source/build-llvm/tools/clang/stage2-stamps/stage2-configure`

Full log:
https://llvm-jenkins.debian.net/job/llvm-toolchain-lunar-binaries/architecture=amd64,distribution=lunar,label=amd64/176/consoleFull

**Comment by joker-eph** [comment]
Seems like stage2 does not disable LLVM_INCLUDE_TESTS? This will result in shipping more stuff than we need I believe.

The log create the library that should produce the symbol here:
```
[9013/10112] : && /usr/bin/cmake -E rm -f lib/libMLIRGPUTestPasses.a && /build/source/build-llvm/bin/llvm-ar Dqc lib/libMLIRGPUTestPasses.a  tools/mlir/test/lib/Dialect/GPU/CMakeFiles/MLIRGPUTestPasses.dir/TestConvertGPUKernelToCubin.cpp.o tools/mlir/test/lib/Dialect/GPU/CMakeFiles/MLIRGPUTestPasses.dir/TestConvertGPUKernelToHsaco.cpp.o tools/mlir/test/lib/Dialect/GPU/CMakeFiles/MLIRGPUTestPasses.dir/TestGpuMemoryPromotion.cpp.o tools/mlir/test/lib/Dialect/GPU/CMakeFiles/MLIRGPUTestPasses.dir/TestGpuRewrite.cpp.o && /build/source/build-llvm/bin/llvm-ranlib -D lib/libMLIRGPUTestPasses.a && :
```

However the file is `TestLowerToNVVM.cpp` and I don't see it here, there is some mystery...

Actually looking further, this was fixed in d661b4b57504b965a37dc30b79bdd5ac36fca9ec and you're just not running at HEAD I believe.
